pkg/adapter/apiserver/ref.go

```diff
@@ -97,21 +92,38 @@ func (a *ref) Delete(obj interface{}) error {
 		return err
 	}
 
-	if _, err := a.ce.Send(context.Background(), *event); err != nil {
-		a.logger.Info("event delivery failed", zap.Error(err))
-		return err
-	}
-	return nil
+	return a.sendEvent(context.Background(), event, a.reporter)
 }
 
 func (a *ref) addControllerWatch(gvr schema.GroupVersionResource) {
+	reportArgs := &ReportArgs{
+		ns:          a.namespace,
+		eventSource: a.source,
+		eventType:   a.eventType,
+	}
+	a.reporter.ReportEventCount(reportArgs, nil)
 	if a.controlledGVRs == nil {
 		a.controlledGVRs = []schema.GroupVersionResource{gvr}
 		return
 	}
 	a.controlledGVRs = append(a.controlledGVRs, gvr)
 }
 
+func (a *ref) sendEvent(ctx context.Context, event *cloudevents.Event, reporter StatsReporter) error {
+	reportArgs := &ReportArgs{
```

Comment:
We will also need to add the importerName and the importerResourceGroup... So that we can also filter based on those attributes. E.g., give me the event_count of all apiserversource importers, and so on...

The name you might need to send it in an env variable to the receive_adapter (as we do in the broker ingress)...
The importerResourceGroup can be hardcoded to `apiserversources.sources.eventing.knative.dev` (put it in a constant)

See labels for tags here: https://github.com/knative/pkg/blob/ec2f20ae67fbbccfac425061f2c7c089d1932105/metrics/metricskey/constants_eventing.go#L52-54

Reply:
Added those tags now. A sample metrics looks like:
```
apiserversource_event_count{event_source="https://1.2.3.4:443",event_type="dev.knative.apiserver.resource.update",**importer_name**="testevents",**importer_resource_group**="apiserversources.sources.eventing.knative.dev",namespace_name="default"} 3
```